filesystem_test.go

```diff
@@ -0,0 +1,240 @@
+package function
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"sort"
+	"testing"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+func TestFileSystems(t *testing.T) {
+	var err error
+
+	tests := []struct {
+		name       string
+		fileSystem Filesystem
+	}{
+		{
+			name:       "embedded",
+			fileSystem: EmbeddedTemplatesFS,
+		},
+		{
+			name:       "os",
+			fileSystem: initOSFS(t),
+		},
+		{
+			name:       "git",
+			fileSystem: initGitFS(t),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			templatesFS := tt.fileSystem
+
+			if templatesFS == nil && runtime.GOOS == "windows" {
+				t.Skip("FS == nil")
```

Comment:
Interestingly this issues doesn't happen when I force `tmp` files not to be created on `C:` drive but on `Z:` drive -- tested on `wine`.

Reply:
That is indeed strange